### ISSUE TITLE
About Separating the training

### ISSUE THREAD

**Issue by JasonSheng-atp** [hzwer/arXiv2020-RIFE#218]
hello, 我尝试修改rife的时候发现整个网络好像比较容易train崩，感觉应该是多个loss的平衡上有些难度。
请问作者有试过分开训练student和teacher网络吗？train 完teacher以后冻住，这样少一个loss会不会对训练有帮助？

**Comment by hzwer** [comment]
你可以把 student 输出加上 detach，与 teacher 截断，影响不大；
我这边 train 崩的情况都能通过两种方式解决
1. 学习率降低到 1e-4；2. 蒸馏 loss 降低权重。

**Comment by JasonSheng-atp** [comment]
![frame2_fake](https://user-images.githubusercontent.com/78075051/144955892-4216fb45-3aa9-4654-b73e-1aa4ccc404d3.png)
![frame2_fake](https://user-images.githubusercontent.com/78075051/144955925-334fb44f-a771-44f9-8c2d-a25e71f3c443.png)
还想请教一下，上一张是RIFE普通版，下一张是RIFE large版，large版包括正向反向计算两次后平均，我发现这个trick确实能提高psnr到36.10左右，但是似乎会导致大运动更加模糊。特别是手部这里。

**Comment by hzwer** [comment]
我感觉 psnr 越高的模型，结果似乎越模糊，参考 distortion-perception trade-offs
https://openaccess.thecvf.com/content_cvpr_2018/papers/Blau_The_Perception-Distortion_Tradeoff_CVPR_2018_paper.pdf
如果要视觉效果好，还是用上 perceptual loss 比较好

**Comment by JasonSheng-atp** [comment]
![frame1](https://user-images.githubusercontent.com/78075051/145328440-1dc82615-a5e9-439c-b2a5-09c86519bde1.png)
![frame3](https://user-images.githubusercontent.com/78075051/145328445-b6779ce7-6e96-4edd-9e0c-913bcc5f10bb.png)
![flow1](https://user-images.githubusercontent.com/78075051/145328450-0feff259-962e-44e4-8367-278469bde5c6.png)
在ATD12K上测试的时候，我还发现了一个问题，就是在前后两帧出现相似物体时，网络有时会生成0光流，破坏了object的整体性，这个会不会和distill_loss会利用pixel相关的mask有关？
**loss_mask = ((merged[i] - gt).abs().mean(1, True) > (merged_teacher - gt).abs().mean(1, True) + 0.01).float().detach()
loss_distill += ((flow_teacher.detach() - flow_list[i]).abs() * loss_mask).mean()**

**Comment by hzwer** [comment]
@JasonSheng-atp 你有把 ATD12K 加入训练么，会不会只是因为泛化不够好

**Comment by JasonSheng-atp** [comment]
我去试试看，这个只用了google drive给的

**Comment by hzwer** [comment]
这张图上看，它只是没把火车看作一个整体，因为火车中间那一截，看起来确实像是没有动。。。

**Comment by JasonSheng-atp** [comment]
我感觉网络在面对**重复纹理**或者有ambiguity的时候会比较困惑，这也是光流的一个普遍难点
